src/objectid.ts

```diff
@@ -73,17 +72,14 @@ export class ObjectId {
       this[kId] = ObjectId.generate(typeof workingId === 'number' ? workingId : undefined);
     } else if (ArrayBuffer.isView(workingId) && workingId.byteLength === 12) {
       // If intstanceof matches we can escape calling ensure buffer in Node.js environments
-      this[kId] = workingId instanceof Buffer ? workingId : ensureBuffer(workingId);
+      this[kId] = ByteUtils.toLocalBufferType(workingId);
     } else if (typeof workingId === 'string') {
       if (workingId.length === 12) {
-        const bytes = Buffer.from(workingId);
-        if (bytes.byteLength === 12) {
-          this[kId] = bytes;
-        } else {
-          throw new BSONTypeError('Argument passed in must be a string of 12 bytes');
-        }
+        throw new BSONError(
+          'TODO, we are not supporting this anymore and the behavior change is not worth fixing'
```

Comment:
I'm breaking this now since I ran into issues with testing but, We can clean it up proper in the follow up ticket to remove the binary strings